rest_framework/routers.py

```diff
@@ -35,6 +35,12 @@
 DynamicListRoute = namedtuple('DynamicListRoute', ['url', 'name', 'initkwargs'])
 
 
+def replace_curly_brackets(url_path):
+    if ('{' and '}') in url_path:
+        url_path = url_path.replace('{', '{{').replace('}', '}}')
+    return url_path
```

Comment:
Error raised at here in routers.py, 258L
```
regex = route.url.format( 
    prefix=prefix, 
    lookup=lookup, 
    trailing_slash=self.trailing_slash 
) 
```
Actually, this escaping is not for regex, for formatting string.
I also check that this replace makes well-routed url with inserted kwargs in url_path in my testcase.